### DISCUSSION TITLE
Use Resign with Match

### DISCUSSION THREAD

**Discussion by PowerOfCreation** [fastlane/fastlane#20576] [Q&A]
Hello, I'm using match together with a Matchfile in a project. I need to do resigning and I would like to use the provisioning profiles that got fetched by calling match. Resign doesn't recognize those values as it only works together with sigh. In the entire documentation it is discouraged to use sigh and it pushes towards the usage of match. Additionally, sigh won't use the Matchfile configuration, which is frustrating. Is there any easy way of using resign in combination with match? It seems to me, that just calling match followed by resign seems to be the most intuitive way and I'm confused why this is not supported. I tried extracting the provisioning profile paths that got outputted by match manually  and supply them to resign, but I'm not very experienced with Swift and couldn't make that work. Help would be much appreciated.

**Comment by ProTip** [comment]
As a workaround Match exports a number of environment variable in the form `sigh_#{AppId}_appstore_#{type}`. You can see these listed in the Match log output.

You can use these types for `resign`:

`certificate-name` => `lane_context[:MY_SIGNING_ID]`

`profile-path` => `lane_context[SharedValues::SIGH_PROFILE_PATH]`

I have a custom `certs` lane that sets the `lane_context` variables after running `match` then calls `resign(signing_identity: lane_context[:MY_SIGNING_ID], ...)`.
